source/bolts/experimental/refraction.d

```diff
@@ -0,0 +1,659 @@
+/++
+    It is sometimes necessary to create a function which is an exact copy of
+    another function. Or sometimes it is necessary to introduce a few
+    variations, while carrying all the other aspects. Because of function
+    attributes, parameter storage classes and user-defined attributes, this
+    requires building a string mixin. In addition, the mixed-in code must refer
+    only to local names, if it is to work across module boundaires.
+
+    This module facilitates the creation of such mixins.
+
+ Copyright: Copyright Jean-Louis Leroy 2017-2020
+
+ License:   $(LINK2 http://boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+ Authors:   Jean-Louis Leroy
++/
+
+module bolts.experimental.refraction;
+
+import std.array;
+import std.format;
+import std.meta;
+import std.algorithm.iteration : map;
+import std.range : iota;
+import std.traits;
+
+template ParameterAttribute(alias F, int i, int j)
+{
+  static if (is(typeof(F) P == __parameters)) {
+    alias ParameterAttribute = Alias!(__traits(getAttributes, P[i..i+1])[j]);
+  }
+}
+
+/**
+    Return a Function object that captures all the aspects of Fun, using the
+    value of localSymbol to represent the return and parameter types, and the
+    UDAs.
+
+    Params:
+    fun = a function
+    localSymbol = a string mixin that represents fun in the caller's context
+
+*/
+
+Function refract(alias fun, string localSymbol)()
```

Comment:
Can `localSymbol` default to `__traits(identifier, fun)`?

Reply:
> Can `localSymbol` default to `__traits(identifier, fun)`?

It's almost always the wrong value. I think it will just mislead users, for what benefit?

Reply:
It seems like it's the correct value from all the tests. Why is it almost always the wrong value?